scripts/variantstore/wdl/GvsImportGenomes.wdl

```diff
@@ -213,31 +211,103 @@ task LoadData {
+
+    ## TODO check if samples even need loading?!? hit the BQ database and get the loaded status for these samples
+
+    # Create temp table with the sample_names and load external sample names into temp table -- make sure it doesn't exist already
+    set +o errexit
+    bq show --project_id ~{project_id} ~{temp_table} > /dev/null
+    BQ_SHOW_RC=$?
+    set -o errexit
+
+    # If there is already a table of sample names or something else is wrong, burn it down to start fresh.
+    if [ $BQ_SHOW_RC -eq 0 ]; then
+      bq rm -t -f --project_id=~{project_id} ~{temp_table}
+    fi
+
+    echo "Creating the external sample name list table ~{temp_table}"
+    bq --project_id=~{project_id} mk ~{temp_table} "sample_name:STRING"
+    NAMES_FILE=~{write_lines(sample_names)}
+    bq load --project_id=~{project_id} ~{temp_table} $NAMES_FILE "sample_name:STRING"
+
+    # Get the current min/max id, or 0 if there are none. Withdrawn samples still have IDs so don't filter them out.
+    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} '
+      SELECT IFNULL(MIN(sample_id),0) as min, IFNULL(MAX(sample_id),0) as max FROM `~{dataset_name}.~{table_name}`
+        AS samples JOIN `~{temp_table}` AS temp ON samples.sample_name = temp.sample_name' > results.csv
+
+    # prep for being able to return min table id
+    min_sample_id=$(tail -1 results.csv | cut -d, -f1)
+    max_sample_id=$(tail -1 results.csv | cut -d, -f2)
+
+    # no samples have been loaded or we don't have the right sample_names or something else is wrong, bail
+    if [ $max_sample_id -eq 0 ]; then
+      echo "Max id is 0. Exiting"
+      exit 1
+    fi
+
+    python3 -c "from math import ceil; print(ceil($max_sample_id/~{samples_per_table}))" > max_sample_id ## TODO do we need this?!?!
+    python3 -c "from math import ceil; print(ceil($min_sample_id/~{samples_per_table}))" > min_sample_id ## TODO do we need this?!?!
+
+    # get sample map of samples that haven't been loaded yet
+    bq --project_id=~{project_id} query --format=csv --use_legacy_sql=false ~{bq_labels} -n ~{num_samples} '
+      SELECT sample_id, samples.sample_name FROM `~{dataset_name}.~{table_name}` AS samples JOIN `~{temp_table}` AS temp ON
+            samples.sample_name = temp.sample_name WHERE
+            samples.sample_id NOT IN (SELECT sample_id FROM `~{dataset_name}.sample_load_status` WHERE status="FINISHED") AND
+            samples.withdrawn is NULL' > sample_map.csv
+
+    cut -d, -f1 sample_map.csv > gvs_ids.csv ## TODO do we need this?!?!
+
+    ## delete the table that was only needed for this ingest test
+    bq --project_id=~{project_id} rm -f=true ~{temp_table}
+
+    ## now we want to create a sub list of these samples (without the ones that have already been loaded)
+    curl https://raw.githubusercontent.com/broadinstitute/gatk/ah_var_store/scripts/variantstore/wdl/extract/curate_input_array_files.py -o curate_input_array_files.py
```

Comment:
I think this might have been suggested at one point, but `CurateInputLists` might be modified to return the curation script as an output, which could then be made an input to this task.